lib/node-token-validation/src/api/TokenValidator.ts

```diff
@@ -195,10 +383,75 @@ export class TokenValidator {
     }
 
     /**
-     * Base function that validates tokens against options provided.
-     * @param token JWT token to be validated
-     * @param {@link (TokenValidationParameters:type)}
-     * @returns {Promise.<TokenValidationResponse>}
+     * Base function for token validation.
+     * 
+     * `validateToken` takes in a JWT token and options in the form of {@link TokenValidationParameters}, which must be set for the token to be validated.
+     * `validateToken` may be called directly, or called by other TokenValidator APIs.
+     * 
+     * The following operations are performed on the token by the JOSE library:
+     * 
+     * - Verifies the JWT format
+     * - Verifies the JWS signature
+     * - Validates the JWT claims set
+     * 
+     * See documentation for more details on JOSE: https://github.com/panva/jose.
+     * 
+     * `validateToken` also performs the following additional operations on the token:
+     * 
+     * - Validates additional claims for id tokens (nonce, c_hash, at_hash)
+     * 
+     * ### Parameters
+     * 
+     * {@link TokenValidationParameters} are passed in and defaults added if not already set.
+     * 
+     * ### Usage example
+     * 
+     * To validate a token using the `validateToken` API, pass in a token along with the {@link TokenValidationParameters}`.
+     * 
+     * ```js
+     * const nodeTokenValidation = require("@azure/node-token-validation");
+     * const tokenValidator = new nodeTokenValidation.TokenValidator();
+     *
+     * const tokenValidationParams = {
+     *    validIssuers: ["issuer-here"],
+     *    validAudiences: ["audience-here"]
+     * };
+     *
+     * tokenValidator.validateToken("token-here", tokenValidationParams).then((response) => {
+     *    // Use token validation response
+     * }).catch((error) => {
+     *    // Handle error here
+     * });
+     * ```
+     * 
+     * A sample demonstrating usage can also be found [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/tree/dev/samples/node-token-validation-samples/basic-sample).
+     * 
+     * ### Returns
+     * 
+     * `validateToken` returns a {@link TokenValidationResponse} if a token is successfully validated.
+     * 
+     * ### Errors
+     * 
+     * The following configuration errors from the Node Token Validation library may be thrown when validating a token.
+     * Ensure the missing token/claims are present before validating.
+     * 
+     * - Missing token error
```

Comment:
Do we want to put the actual error codes here?

Reply:
Oops, missed this comment. Yes, that's a great idea. Added the error codes.